unified-runtime/source/adapters/level_zero/v2/queue_api.hpp

```diff
@@ -27,6 +27,12 @@ struct ur_queue_t_ {
                                            ur_native_handle_t *) = 0;
   virtual ur_result_t queueFinish() = 0;
   virtual ur_result_t queueFlush() = 0;
+
+  virtual ur_result_t
+  runBatchIfActive([[maybe_unused]] int64_t batch_generation) {
```

Comment:
`onEventWaitListUse` it conveys what is happening, and what the queue should react to, in a more generic way. A batched queue will just run the batch "on event use inside of a wait list".